Automatically determine the last modification date of Markdown files from git

tsibley opened the pull request:
This frees us from having to remember to bump the date in the frontmatter when we make changes.

See each commit message for rationale.

tsibley committed:
Automatically determine the last modification date of Markdown files from git

This frees us from having to remember to bump the date in the
frontmatter when we make changes.

Only documents with no "date" field in the frontmatter will use the date
from git.  This allows some documents to set a fixed explicit date, such as
"February 2018", if they so desire.  Blog posts and reports, for
example, may want the timestamp to ignore minor updates made after their
initial posting.

tsibley committed:
Remove "date" from frontmatter for about/ and docs/ pages

This lets their dates come from git, which frees us from remembering to
bump the date when we make changes.  (See the previous commit for how.)

I left the dates for blog/ and reports/ untouched, as those likely want
to ignore minor updates and use their initial post date (and in some
cases they use date spans like "February 2018").

tsibley committed:
Distinguish post dates from last modified dates in the page heading

Some pages used the date to indicate when the initial posting was made,
others used it to indicate when a page was last updated.  Since the
latter usage is now done automatically, we can infer that explicit dates
are only "post dates" and omit the indication that they're a "Last
modified" timestamp.

jameshadfield reviewed (approved):
Seems good to me.

Would you please consider adding linting to your workflow?

tsibley committed:
Move eslint ignore patterns into a config file

That way they're respected no matter how eslint is run (e.g. `npm run
lint:js` vs `eslint`).

Unfortunately there is no config file equivalent of --ext.

tsibley commented:
Thanks for looking! I re-pushed the branch to address linting warnings for the code I touched and merged the result.

Sorry I keep forgetting the linting. I have rarely used _syntax_ linters in the past because of fundamental beliefs about their usefulness. I'm willing to adopt my ways here since they're already being used though! However, it would be helpful to me (and probably more broadly) if we could work `master` down to zero linting errors and then add it to the Travis testing. That would help flag warnings even if someone (like me!) forgets to run eslint locally before pushing.
